pkg/providers/instance/instance.go

```diff
@@ -424,6 +448,30 @@ func (p *DefaultProvider) isMixedCapacityLaunch(nodeClaim *corev1beta1.NodeClaim
 	return hasSpotOfferings && hasODOffering
 }
 
+// isMixedCapacityLaunch returns true if nodepools and available offerings could potentially allow either a spot or
+// and on-demand node to launch
+func (p *DefaultProvider) isCapacityReservationLaunch(nodeClaim *corev1beta1.NodeClaim, instanceTypes []*cloudprovider.InstanceType) bool {
+	requirements := scheduling.NewNodeSelectorRequirementsWithMinValues(nodeClaim.Spec.Requirements...)
+	// requirements must allow on-demand
+	if !requirements.Get(corev1beta1.CapacityTypeLabelKey).Has(corev1beta1.CapacityTypeOnDemand) {
+		return false
+	}
+
+	requirements[corev1beta1.CapacityTypeLabelKey] = scheduling.NewRequirement(corev1beta1.CapacityTypeLabelKey, v1.NodeSelectorOpIn, corev1beta1.CapacityTypeOnDemand)
+	requirements[v1beta1.LabelCapactiyReservationID] = scheduling.NewRequirement(v1beta1.LabelCapactiyReservationID, v1.NodeSelectorOpExists)
+	for _, instanceType := range instanceTypes {
+		for _, offering := range instanceType.Offerings {
+			if requirements.Compatible(offering.Requirements, scheduling.AllowUndefinedWellKnownLabels) == nil {
```

Comment:
I think we just need to ensure that the instance type has a compatible offering. That can be done just by doing the `HasCompatible()` function on the offerings and then creating an `Exists` requirement for the ODCR ID key to filter out any instance types where there is no offering that has requirements that contain that key (no ODCR offering available)